assembly/src/assembler/instruction/u32_ops.rs

```diff
@@ -518,8 +384,39 @@ fn handle_division(
         _ => unreachable!("unsupported operation mode"),
     }
 
-    span.push_op(U32div);
+    span.add_op(U32div)
+}
+
+// BITWISE OPERATIONS - HELPERS
+// ================================================================================================
 
+/// Prepare a bitwise operation argument, extracting it either from the stack, or from the provided
+/// immediate value.
+///
+/// This function supports only checked and unchecked modes; if some other mode is provided, it
+/// will panic.
+fn prepare_bitwise(
+    span: &mut SpanBuilder,
+    imm: Option<u8>,
+    op_mode: U32OpMode,
+) -> Result<(), AssemblyError> {
+    match (imm, op_mode) {
+        (Some(imm), U32OpMode::Checked) => {
+            validate_param(imm, 0, 31)?;
```

Comment:
Instead of hard-coding `31` here, I'd prefer to use `MAX_U32_SHIFT_VALUE`/`MAX_U32_ROTATE_VALUE` constants defined in `lib.rs`. Since we don't know which operation we are performing here, maybe we should combine these constants into one. Maybe something like: `MAX_U32_BITWISE_IMM_VALUE`.

Reply:
Fixed